rbac/management/principal/view.py

```diff
@@ -243,3 +210,39 @@ def users_from_proxy(self, user, query_params, options, limit, offset):
             org_id=user.org_id, input=proxyInput, limit=limit, offset=offset, options=options
         )
         return resp, ""
+
+    @staticmethod
+    def service_accounts_from_it_service(request, user, query_params, options):
+        """Format Service Account request for IT Service and return prepped result."""
+        options["email"] = query_params.get(EMAIL_KEY)
+        options["match_criteria"] = validate_and_get_key(
+            query_params, MATCH_CRITERIA_KEY, VALID_MATCH_VALUE, required=False
+        )
+        options["username_only"] = validate_and_get_key(
+            query_params, USERNAME_ONLY_KEY, VALID_BOOLEAN_VALUE, required=False
+        )
+        options["usernames"] = query_params.get(USERNAMES_KEY)
+
+        # Fetch the service accounts from IT.
+        token_validator = ITSSOTokenValidator()
+        user.bearer_token = token_validator.validate_token(
+            request=request, additional_scopes_to_validate=set[ScopeClaims]([ScopeClaims.SERVICE_ACCOUNTS_CLAIM])
+        )
+
+        try:
+            it_service = ITService()
+            service_accounts, sa_count = it_service.get_service_accounts(user=user, options=options)
+        except (requests.exceptions.ConnectionError, UnexpectedStatusCodeFromITError):
+            return Response(
```

Comment:
If there is an error, I think this response will be stored in the resp, where it is called, and process normally instead of returning the response. Maybe we can try catch outside of this method service_accounts_from_it_service

Reply:
yeah that is a good point ... it looks like in this case we should return dict with error message and then [here](https://github.com/RedHatInsights/insights-rbac/blob/a8c204fae160251baf78648b1dee0abba6c56824/rbac/management/principal/view.py#L176) is decided if the response contains 200 or something else ... it is solved the same way for user based principals when BOP is called ... I will add a test for this case